src/Kentico.Xperience.AzureSearch/Indexing/DefaultAzureSearchClient.cs

```diff
@@ -154,16 +154,18 @@ private async Task RebuildInternal(AzureSearchIndex azureSearchIndex, Cancellati
                     foreach (var contentType in includedPathAttribute.ContentTypes)
                     {
                         queryBuilder.ForContentType(contentType.ContentTypeName, config => config.ForWebsite(azureSearchIndex.WebSiteChannelName, includeUrlPath: true, pathMatch: pathMatch));
```

Comment:
Why don't we use the [ForContentTypes](https://docs.kentico.com/documentation/developers-and-admins/api/content-item-api/reference-content-item-query#forcontenttypes-parameterization) parametrization? We can specify OfContentTypes and also ForWebsite. The benefit of this is that ForContentTypes does not load the content item data and therefore we do not make multiple queries inside for each content type. As we do not use the specific content item data, this will have significant performance improvement. 
Ideally we could migrate to Content retriever https://docs.kentico.com/documentation/developers-and-admins/api/content-item-api/content-retriever-api

Reply:
Same for reusable items bellow.